src/Microsoft.VisualStudio.Services.Agent/Capabilities/EnvironmentCapabilitiesProvider.cs

```diff
@@ -84,11 +87,29 @@ public Task<List<Capability>> GetCapabilitiesAsync(AgentSettings settings, Cance
                     continue;
                 }
 
+                if (isKeySecretEnvironmentVariable(name, secretKnobs))
+                {
+                    HostContext.SecretMasker.AddValue(value);
+                    HostContext.SecretMasker.MaskSecrets(value);
+                }
                 Trace.Info($"Adding '{name}': '{value}'");
                 capabilities.Add(new Capability(name, value));
             }
 
             return Task.FromResult(capabilities);
         }
+
+        private bool isKeySecretEnvironmentVariable(string name, IEnumerable<Knob> secretKnobs)
```

Comment:
Could you please name it more specifically? This method only checks if variable is related to one of the secret knobs, although it does not determine if variable is secret generally